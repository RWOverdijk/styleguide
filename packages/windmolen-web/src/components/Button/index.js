// @flow
import type { Node } from 'react';
import React from 'react';
import Pressable, { type VariantType } from '../Pressable';

export type Props = {
  /** The actual icon to use, same as <Icon /> */
  icon?: string,
  hideArrow?: boolean,
  children?: Node,
  small?: boolean,
  variant?: VariantType
};

const Button = (props: Props) => {
  // TODO wait for do expressions to land in ecmascript, because i will NOT do `let`
  const fontSize = (function(props) {
    if (props.small) {
      return 'button-small';
    }

    if (props.variant === 'text') {
      return 'inherit';
    }

    return 'button';
  })(props);

  return <Pressable fontSize={fontSize} {...props} />;
};

Button.defaultProps = {
  as: 'button',
  variant: 'button-primary',
  hideArrow: false,
};

export default Button;
