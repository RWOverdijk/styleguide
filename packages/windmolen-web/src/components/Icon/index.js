// @flow
import type { Node } from 'react';
import React from 'react';
import styled from 'styled-components';
import Span from '../Span';
import SpritesheetSVG from '../../assets/icons/sprite-sheet.svg';
import SpritesheetPNG from '../../assets/icons/sprite-sheet.png';
import { colors } from '../../globals';

type Props = {
  name: string,
  fontSize: string,
  variant?: number,
};

const spritesheet = {};

// Define the amount of rows (1-index based).
spritesheet.rows = 5;

// Define the icons from left to right in the sheet using a human-readable name.
spritesheet.names = [
  'facebook',
  'twitter',
  'youtube',
  'instagram',
  'arrow-left',
  'arrow-right',
  'search',
  'thumb-up',
  'app',
  'laadpas',
  'laadpaal',
  'slim-laden',
  'solar-panel',
  'windmill',
  'bio',
  'send',
  'sun',
  'user',
  'phone',
  'app--left',
  'laadpas--left',
  'laadpaal--left',
  'slim-laden--left',
  'leaf',
  'windmill-blades',
  'caret-right',
  'star',
  'play',
  'pause',
  'enter-fullscreen',
  'exit-fullscreen',
  'volume-muted',
  'volume-low',
  'volume-mid',
  'volume-low',
  'wind-indication',
  'star-outline--thick',
  'vdb-logo',
  'wallet--thick',
  'meter--thick',
  'switch--thick',
  'solarpanel--thick',
  'thunderstrike--thick',
  'user--thick',
  'paper--thick',
  'label--thick',
  'bell--thick',
  'lamp--thick',
  'meter-switch--thick',
  'laadpas--thick',
  'laadpaal--thick',
  'app--thick',
  'thunderstrike',
  'wallet',
  'repair',
  'battery',
  'laptop',
  'calendar',
  'notes',
  'phone-thin',
  'gas',
  'apartment',
  'house--tussenwoning',
  'house--hoekwoning',
  'house--2-onder-1-kap',
  'house--vrijstaand',
  'people--one',
  'people--two',
  'people--three',
  'people--four',
  'people--five',
  'electricity-gas',
  'apartment--thick',
  'house--tussenwoning--thick',
  'house--hoekwoning--thick',
  'house--2-onder-1-kap--thick',
  'house--vrijstaand--thick',
  'people--one--thick',
  'people--two--thick',
  'people--three--thick',
  'people--four--thick',
  'people--five--thick',
  'electricity-gas--thick',
  'radio',
  'radio--checked',
  'checkbox',
  'checkbox--checked',
  'checkmark',
];

const StyledSpan = styled(Span)`
  background-image: url('${SpritesheetPNG}');
  background-image: url('${SpritesheetSVG}');
  background-position: -${props => spritesheet.names.indexOf(props.name)}em -${props => props.variant}em;
  background-repeat: no-repeat;
  background-color: ${colors.transparent};
  background-size: ${spritesheet.names.length}em ${spritesheet.rows}em;
  display: inline-block;
  height: 1em;
  width: 1em;
  vertical-align: middle;

  &&& {
    font-size: ${props => props.fontSize};
    line-height: 1;
  }
}`;

const Icon = (props: Props): Node => (
  <StyledSpan {...props} />
);

Icon.defaultProps = {
  fontSize: '150%',
  variant: 0,
};

export {
  spritesheet,
};

export default Icon;
