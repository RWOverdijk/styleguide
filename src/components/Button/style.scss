/** @define Button */

@import "../../css/_global.scss";
@import "../../css/_mixins";

.Van-Button {
  display: inline-flex;
  align-items: center;
  font-size: 18px;
  justify-content: center;
  text-align: center;
  white-space: nowrap;
  padding: 0 15px;
  margin: 0;
  cursor: pointer;
  transition: background-color 300ms ease-in-out, border-color 300ms ease-in-out;
  border-radius: 5px;
  min-height: 50px;
  border: 2px solid $color-grey-bright;
  background-color: $color-grey-bright;

  // Colors
  &--primary {
    @include background-color($color-white, $color-pink-primary);

    &.Van-Button--inverted { @include background-color($color-white, $color-pink-primary, true); }
  }

  &--info {
    @include background-color($color-white, $color-blue-primary);

    &.Van-Button--inverted { @include background-color($color-white, $color-blue-primary, true); }
  }

  &--warning {
    @include background-color($color-white, $color-orange);

    &.Van-Button--inverted { @include background-color($color-white, $color-orange, true); }
  }

  &--danger {
    @include background-color($color-white, $color-red);

    &.Van-Button--inverted { @include background-color($color-white, $color-red, true); }
  }

  &--success {
    @include background-color($color-white, $color-green-primary);

    &.Van-Button--inverted { @include background-color($color-white, $color-green-primary, true); }
  }

  // Sizes
  &--large {
    min-height: 60px;
    font-size: 20px;
    line-height: 24px;
  }

  &--small {
    min-height: 40px;
    font-size: 18px;
    line-height: 22px;
  }

  &--tiny {
    min-height: 35px;
    font-size: 16px;
    line-height: 19px;
  }

  &--compact {
    min-height: 26px;
    font-size: 13px;
    line-height: 16px;
  }

  &--full {
    width: 100%;
  }

  > .Van-Icon {
    /* postcss-bem-linter: ignore */
    &:first-child {
      margin-right: 0.5rem;
    }

    /* postcss-bem-linter: ignore */
    &:last-child {
      margin-left: 0.5rem;
    }
  }
}
