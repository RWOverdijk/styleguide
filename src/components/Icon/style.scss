.Van-Icon {
  display: inline-block;
  width: 20px;
  height: 20px;
  vertical-align: middle;

  > svg {
    position: relative;
    z-index: 2;
    display: block;
    width: 100%;
    height: 100%;
  }
}
