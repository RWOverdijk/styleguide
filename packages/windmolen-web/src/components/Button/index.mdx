---
name: Button
menu: Controls
route: /Controls/Button
---

import { Playground, PropsTable } from 'docz'
import Button from './'
import { colors } from '../../globals';
import windmill from '../../assets/icons/windmill.svg';

# Button

## Props
<PropsTable of={Button} />

## Primary
<Playground>
  <Button>Primary</Button>
</Playground>

## Secondary
<Playground style={{ backgroundColor: colors.charcoalGray }}>
  <Button type="secondary">Secondary</Button>
</Playground>

## Small
<Playground>
  <Button small>Small</Button>
</Playground>

## With icon
<Playground>
  <Button withIcon={windmill}>With icon</Button>
</Playground>

## Text button
<Playground>
  <Button asTextButton>Text button</Button>
</Playground>
