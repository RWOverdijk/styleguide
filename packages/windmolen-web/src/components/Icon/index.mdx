---
name: Icon
route: /Elements/Icon
menu: Elements
---

import { Playground, PropsTable } from 'docz';
import Icon from './';
import windmill from '../../assets/icons/windmill.svg';

# Icon

## Props
<PropsTable of={Icon} />

## Basic usage
<Playground>
  <Icon source={windmill} />
</Playground>
