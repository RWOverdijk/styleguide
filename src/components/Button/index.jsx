import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class Button extends PureComponent {
  render() {
    const {
      className,
      type,
      size,
      full,
      inverted,
      children,
      ...props
    } = this.props;

    return (
      <button
        className={classnames('Van-Button', className, {
          'Van-Button--primary': type === 'primary',
          'Van-Button--info': type === 'info',
          'Van-Button--warning': type === 'warning',
          'Van-Button--danger': type === 'danger',
          'Van-Button--success': type === 'success',

          'Van-Button--small': size === 'small',
          'Van-Button--compact': size === 'compact',
          'Van-Button--large': size === 'large',
          'Van-Button--tiny': size === 'tiny',

          'Van-Button--full': full,

          'Van-Button--inverted': inverted,
        })}
        {...props}
      >
        {children instanceof Array ? children.map(c => (
          (typeof c === 'string') ? <span key={c}>{c}</span> : c
        )) : children}
      </button>
    );
  }
}

Button.defaultProps = {
  className: '',
  type: '',
  size: '',
  full: false,
  inverted: false,
  children: '',
};

Button.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  size: PropTypes.string,
  full: PropTypes.bool,
  inverted: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Button;
