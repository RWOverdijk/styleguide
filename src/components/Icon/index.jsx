/* eslint-disable react/no-danger */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import isSVG from 'is-svg';

import './style.scss';

class Icon extends PureComponent {
  render() {
    const { className, source, ...props } = this.props;

    if (!isSVG(source)) {
      return null;
    }

    return (
      <span
        className={classnames('Van-Icon', className)}
        dangerouslySetInnerHTML={{ __html: source }}
        {...props}
      />
    );
  }
}

Icon.defaultProps = {
  className: '',
  source: ''
};

Icon.propTypes = {
  className: PropTypes.string,
  source: PropTypes.string
};

export default Icon;
