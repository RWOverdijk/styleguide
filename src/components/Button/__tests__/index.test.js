import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../../Icon';
import Button from '../';

describe("<Button>", () => {
  it("renders primary button type if does not pass a type", () => {
    const tree = renderer.create(
      <Button>My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders primary button type", () => {
    const tree = renderer.create(
      <Button type="primary">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders info button type", () => {
    const tree = renderer.create(
      <Button type="info">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders warning button type", () => {
    const tree = renderer.create(
      <Button type="warning">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders danger button type", () => {
    const tree = renderer.create(
      <Button type="danger">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders success button type", () => {
    const tree = renderer.create(
      <Button type="success">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders small button size", () => {
    const tree = renderer.create(
      <Button size="small">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders compact button size", () => {
    const tree = renderer.create(
      <Button size="compact">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders large button size", () => {
    const tree = renderer.create(
      <Button size="large">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders tiny button size", () => {
    const tree = renderer.create(
      <Button size="tiny">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders full button", () => {
    const tree = renderer.create(
      <Button full>My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders inverted button", () => {
    const tree = renderer.create(
      <Button inverted>My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Button className="my-class">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Button data-id="0">My button</Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with button on the right", () => {
    const tree = renderer.create(
      <Button>
        My button
        <Icon source="facebook" />
      </Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with button on the left", () => {
    const tree = renderer.create(
      <Button>
        <Icon source="facebook" />
        My button
      </Button>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
