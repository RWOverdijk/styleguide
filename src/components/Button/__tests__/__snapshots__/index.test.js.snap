// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders compact button size 1`] = `
<button
  className="Van-Button Van-Button--primary Van-Button--compact"
>
  My button
</button>
`;

exports[`<Button> renders danger button type 1`] = `
<button
  className="Van-Button Van-Button--danger"
>
  My button
</button>
`;

exports[`<Button> renders full button 1`] = `
<button
  className="Van-Button Van-Button--primary Van-Button--full"
>
  My button
</button>
`;

exports[`<Button> renders info button type 1`] = `
<button
  className="Van-Button Van-Button--info"
>
  My button
</button>
`;

exports[`<Button> renders inverted button 1`] = `
<button
  className="Van-Button Van-Button--primary Van-Button--inverted"
>
  My button
</button>
`;

exports[`<Button> renders large button size 1`] = `
<button
  className="Van-Button Van-Button--primary Van-Button--large"
>
  My button
</button>
`;

exports[`<Button> renders primary button type 1`] = `
<button
  className="Van-Button Van-Button--primary"
>
  My button
</button>
`;

exports[`<Button> renders primary button type if does not pass a type 1`] = `
<button
  className="Van-Button Van-Button--primary"
>
  My button
</button>
`;

exports[`<Button> renders small button size 1`] = `
<button
  className="Van-Button Van-Button--primary Van-Button--small"
>
  My button
</button>
`;

exports[`<Button> renders success button type 1`] = `
<button
  className="Van-Button Van-Button--success"
>
  My button
</button>
`;

exports[`<Button> renders tiny button size 1`] = `
<button
  className="Van-Button Van-Button--primary Van-Button--tiny"
>
  My button
</button>
`;

exports[`<Button> renders warning button type 1`] = `
<button
  className="Van-Button Van-Button--warning"
>
  My button
</button>
`;

exports[`<Button> renders with button on the left 1`] = `
<button
  className="Van-Button Van-Button--primary"
>
  <span
    className="Van-Icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test-file-stub",
      }
    }
    source="facebook"
  />
  <span>
    My button
  </span>
</button>
`;

exports[`<Button> renders with button on the right 1`] = `
<button
  className="Van-Button Van-Button--primary"
>
  <span>
    My button
  </span>
  <span
    className="Van-Icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test-file-stub",
      }
    }
    source="facebook"
  />
</button>
`;

exports[`<Button> renders with specific className 1`] = `
<button
  className="Van-Button my-class Van-Button--primary"
>
  My button
</button>
`;

exports[`<Button> renders with specific className 2`] = `
<button
  className="Van-Button Van-Button--primary"
  data-id="0"
>
  My button
</button>
`;
